### PR TITLE
Install add-on discovery file from openhab-addons

### PR DESCRIPTION
**Pull request by J-N-K**
Related to https://github.com/openhab/openhab-core/pull/3865

### CODE REVIEW

**Comment by J-N-K** [comment]
@andrewfg The aggregated file is copied to `<userdata>/addons`. You should read all files from that folder, we can then easily add other XMLs (e.g. for zwave and zigbee).

File: features/distro/src/main/feature/feature.xml [hunk 1]
```diff
@@ -9,4 +9,8 @@
         <bundle>mvn:org.openhab.ui.bundles/org.openhab.ui.iconset.classic/${project.version}</bundle>
     </feature>
 
+    <feature name="openhab-addon-discovery" description="Discovery Information" version="${project.version}">
+        <configfile finalname="${openhab.userdata}/addons/openhab-addons.xml" override="false">mvn:org.openhab.addons.features.karaf/org.openhab.addons.features.karaf.openhab-addons-external/4.1.0-SNAPSHOT/xml/addons</configfile>
```
Comment (kaikreuzer) [inline features/distro/src/main/feature/feature.xml:13]: Do we have to hardcode the version in here or could we use `${project.version}` as well?

File: distributions/openhab/pom.xml [hunk 2]
```diff
@@ -170,6 +170,7 @@
                         <feature>system</feature>
                         <feature>openhab-runtime-base</feature>
                         <feature>openhab-runtime-ui</feature>
+                        <feature>openhab-addons-discovery</feature>
```
Comment (andrewfg) [inline distributions/openhab/pom.xml:173]: Should this be `openhab-addon-discovery` without "s" ??

File: distributions/openhab/src/main/filtered-resources/userdata/etc/org.apache.karaf.features.cfg [hunk 3]
```diff
@@ -44,6 +44,7 @@ featuresBoot = \
     jaas/${karaf.version}, \
     openhab-runtime-base, \
     openhab-runtime-ui, \
+    openhab-addon-discovery, \
```
Comment (andrewfg) [inline distributions/openhab/src/main/filtered-resources/userdata/etc/org.apache.karaf.features.cfg:47]: Or with an "s"

File: features/distro/src/main/feature/feature.xml [hunk 4]
```diff
@@ -9,4 +9,8 @@
         <bundle>mvn:org.openhab.ui.bundles/org.openhab.ui.iconset.classic/${project.version}</bundle>
     </feature>
 
+    <feature name="openhab-addon-discovery" description="Discovery Information" version="${project.version}">
```
Comment (andrewfg) [inline features/distro/src/main/feature/feature.xml:12]: Ditto

**Review by andrewfg — CHANGES_REQUESTED** [review]

File: features/distro/src/main/feature/feature.xml [hunk 5]
```diff
@@ -9,4 +9,8 @@
...
+        <configfile finalname="${openhab.userdata}/addons/openhab-addons.xml" override="false">mvn:org.openhab.addons.features.karaf/org.openhab.addons.features.karaf.openhab-addons-external/${project.version}/xml/addons</configfile>
```
Comment (andrewfg) [inline features/distro/src/main/feature/feature.xml:13]: Just checking: does the source file name here, match the source file name in https://github.com/openhab/openhab-addons/blob/a9a79b09bf5473bcdd08717f043f8ddfe48fbc6a/features/openhab-addons-external/pom.xml#L35C1-L35C1 ??

File: distributions/openhab/pom.xml (hunk 2 above)
Comment (J-N-K) [reply distributions/openhab/pom.xml]: I decided to use `addon`. But it doesn't matter as long as they are all the same.

File: features/distro/src/main/feature/feature.xml (hunk 5 above)
Comment (J-N-K) [reply features/distro/src/main/feature/feature.xml:13]: Correct.

File: distributions/openhab/pom.xml [hunk 6]
```diff
@@ -170,6 +170,7 @@
                         <feature>system</feature>
                         <feature>openhab-runtime-base</feature>
                         <feature>openhab-runtime-ui</feature>
+                        <feature>openhab-addon-discovery</feature>
```
Comment (andrewfg) [inline distributions/openhab/pom.xml:173]: LGTM

**Review by andrewfg — APPROVED** [review]

**Comment by mherwege** [comment]
LGTM

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Comment by andrewfg** [comment]
@J-N-K I think this PR should now build. Or??

**Comment by kaikreuzer** [comment]
Yes, it does - I only waited with the merge until the build went green and then forgot about it. 🤦

**Pull request merged**